osf/models/user.py

```diff
@@ -456,10 +457,17 @@ def social_links(self):
         social_user_fields = {}
         for key, val in self.social.items():
             if val and key in self.SOCIAL_FIELDS:
-                if not isinstance(val, basestring):
-                    social_user_fields[key] = val
+                if isinstance(self.SOCIAL_FIELDS[key], basestring):
+                    if isinstance(val, basestring):
+                        social_user_fields[key] = self.SOCIAL_FIELDS[key].format(val)
+                    else:
+                        # Provide a single account url for services which we allow multiple-accounts
+                        social_user_fields[key] = self.SOCIAL_FIELDS[key].format(val[0])
                 else:
-                    social_user_fields[key] = self.SOCIAL_FIELDS[key].format(val)
+                    if not isinstance(val, basestring):
+                        social_user_fields[key] = val
+                    else:
+                        social_user_fields[key] = [val]
```

Comment:
Is this a possible state, currently?

Reply:
The first one (The value should be an array and is an array in self.social) is with profile websites. The latter (The value should be an array but is an a string in self.social) is not a current existing case.